app/controllers/posts_controller.rb

```diff
@@ -1,102 +1,74 @@
   load_permissions_and_authorize_resource
-  before_action :set_council
+  load_and_authorize_resource :council, parent: true, find_by: :url
+
+
+  before_action :set_permissions
+  before_action :set_councils, only: [:new, :edit, :update, :create]
 
   def remove_profile
-    @profile = Profile.find_by_id(params[:profile_id])
-    @post.profiles.delete(@profile)
-    respond_to do |format|
-      format.html { redirect_to council_posts_path(@council), notice: @profile.name.to_s + ' har inte längre posten ' + @post.title.to_s + '.' }
-    end
+    profile = Profile.find_by_id(params[:profile_id])
+    @post.remove_profile(profile)
+    redirect_to council_posts_path(@council),
+                notice: %(#{profile.full_name} har inte längre posten #{@post.title}.)
   end
 
   def add_profile_username
     @user = User.find_by(username: params[:username])
-    if @user != nil
-      @profile = @user.profile
-    end
-    if @profile == nil
-      respond_to do |format|
-        format.html { redirect_to council_posts_path(@council), flash: {alert: 'Hittade ingen användare med det användarnamnet.'} }
-      end
-    elsif @profile.name.blank?
-      redirect_to council_posts_path(@council), flash: {alert: 'Användaren :"' + @user.username.to_s + '" måste fylla i fler uppgifter i sin profil.'}
+    if @post.add_profile(@user)
+      redirect_to council_posts_path(@council),
+                  flash: {alert: 'Hittade ingen användare med det användarnamnet.'}
     elsif @profile.posts.include?(@post)
-      redirect_to council_posts_path(@council), flash: {alert: @profile.name.to_s + '(' + @user.username.to_s + ') har redan posten '+@post.title.to_s + '.'}
+      redirect_to council_posts_path(@council),
+                  alert: %(#{@profile.name} ( #{@user.username}) har redan posten #{@post.title} .)
     elsif (@post.limit != nil) && (@post.profiles.size >= @post.limit)
-      redirect_to council_posts_path(@council), flash: {alert: @post.title.to_s + ' har sitt maxantal.'}
+      redirect_to council_posts_path(@council),
+                  alert: %(#{@post.title} har sitt maxantal.)
     else
       @post.profiles << @profile
-      redirect_to council_posts_path(@council), notice: @profile.name.to_s + ' (' + @profile.user.username.to_s + ') tilldelades posten '+@post.title.to_s + '.'
-      if (@profile.first_post == nil)
-        @profile.update(first_post: @post.id)
-      end
+      redirect_to council_posts_path(@council),
+                  notice: %(#{@profile.name} ( #{@profile.user.username}) tilldelades posten #{@post.title}.)
     end
   end
 
   def index
-    if (@council)
-      @posts = @council.posts
-    else
-      @posts = Post.all
-    end
+    @posts = (@council.present?) ? @council.posts : Post.all
     @post_grid = initialize_grid(@posts)
   end
 
   def new
     @post = @council.posts.build
-    @councils = Council.order(title: :asc)
+    @post_permissions = @post.permissions.collect! { |p| p.id }
```

Comment:
1. Use ``permissions = @post.permissions.map &:id

2. No need to do this in the controller, do it when you need the values instead (in the view). Also then you don't have to put it in an instance variable. In general try to set only a single instance variable in the controller.